backend/src/main/kotlin/org/loculus/backend/service/submission/FileMappingPreconditionValidator.kt

```diff
@@ -44,15 +44,25 @@ class FileMappingPreconditionValidator(
         return validateCategoriesMatchSchema(fileCategoriesFilesMap, allowedFileCategories, organism)
     }
 
-    fun validateCategoriesMatchProcessedSchema(
+    fun validateCategoriesMatchOutputSchema(
         fileCategoriesFilesMap: FileCategoryFilesMap?,
         organism: Organism,
     ): FileMappingPreconditionValidator {
         if (fileCategoriesFilesMap == null) return this
-        val allowedFileCategories = backendConfig
+        val allowedCategories = backendConfig
             .getInstanceConfig(organism)
             .schema.files
-        return validateCategoriesMatchSchema(fileCategoriesFilesMap, allowedFileCategories, organism)
+            .map { it.name }
+            .toSet()
+
+        fileCategoriesFilesMap.categories.forEach { category: FileCategory ->
+            if (!allowedCategories.contains(category)) {
+                throw UnprocessableEntityException(
+                    "The category $category is not part of the configured output categories for ${organism.name}.",
+                )
+            }
+        }
+        return this
```

Comment:
Sorry if I'm being blind but isn't that exactly what `validateCategoriesMatchSchema()` is doing? What's different?